tests/timetables/test_interval_timetable.py

```diff
@@ -253,3 +253,83 @@ def test_cron_next_dagrun_info_alignment(last_data_interval: DataInterval, expec
         restriction=TimeRestriction(None, None, True),
     )
     assert info == expected_info
+
+
+def test_cron_interval_dst_next_entering():
+    """
+    This test is composed of several tests targeting an interval
+    timetable crossing a timezone change boundary, e.g. DST
+    """
+    # Interval cron
+    cron_interval = CronDataIntervalTimetable(
+        "0 9-16 * * 1-5", timezone=pendulum.timezone("America/New_York")
+    )
+    # Fixed cron
+    cron_fixed = CronDataIntervalTimetable("0 9 * * 1-5", timezone=pendulum.timezone("America/New_York"))
```

Comment:
This and the test below need to get rid of 1-5 so that they're fixed not interval